deltachat-ffi/src/lib.rs

```diff
@@ -2390,11 +2391,13 @@ pub unsafe extern "C" fn dc_chat_get_info_json(
     }
     let ffi_context = &*context;
     ffi_context
-        .with_inner(|ctx| match chat::get_info_json(ctx, chat_id) {
-            Ok(s) => s.strdup(),
+        .with_inner(|ctx| match deltachat::json::chat_get_info(ctx, chat_id) {
+            Ok(data) => serde_json::to_string(&data)
+                .unwrap_or_log_default(ctx, "dc_get_chat_info_json")
+                .strdup(),
             Err(err) => {
-                error!(ctx, "get_info_json({}) returned: {}", chat_id, err);
-                return "".strdup();
+                ffi_context.error(&format!("dc_get_chat_info_json: {}", err));
```

Comment:
This really isn't any better or worse then the original way.  I just re-wrote this too many times and didn't know what the orignal was anymore.  In general things could be made more ergonomic in this module, but that's for another day.